### PR TITLE
Making TI_ONLY display mode visible to TI Members not TI clients

### LINKED ISSUE #4857: Applicant who is managed by a TI can see TI_ONLY programs
**Describe the bug**
[This](https://github.com/civiform/civiform/blob/39b6db15f773f918d964b32aa68d16edd2f82079/server/app/services/applicant/ApplicantService.java#L849C99-L851) uses the wrong function to determine if an account is a TI account. This kind of works, because when a TI is view the index page as one of their clients, `getManagedByGroup` is going to be present since the TI is managing them.  However, this means the applicant can then log in and see that same program that is supposed to be visible only to TIs.  Instead, this should be using `getMemberOfGroup`, which indicates an account is a member of a TI organization.

**To Reproduce**

1. Create a program with TI visibility.
2. Create an applicant as TI (give the account an email of something@example.com, since our dev environment always adds @example.com to logins).
3. Note that when applying as this applicant, you can see the program.
4. Log out.
5. Log in as that account (something@example.com). Note that you can still see the program.

**Expected behavior**
A TI, viewing the index page as an applicant, can see the TI_ONLY program.  An applicant managed by a TI, when logging in on their own, can not see it unless they've applied to it.

### PR DESCRIPTION
**Pull request by swatkat1**
### Description

The TI_ONLY program visibility condition displayed the program to TI Clients who could create an account with the email address shared with the TI. This PR fixes the bug. 

## Release notes
Applicants who are managed by a TI cannot see TI_ONLY programs

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/helper/feature-flags.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://docs.civiform.us/contributor-guide/developer-guide/feature-flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #4857

### CODE REVIEW

File: server/app/services/applicant/ApplicantService.java [hunk 1]
```diff
@@ -834,39 +834,33 @@ public ImmutableList<Application> getApplications(TimeFilter submitTimeFilter) {
...
-  public CompletionStage<ApplicationPrograms> relevantProgramsForApplicant(long applicantId) {
+  public CompletionStage<ApplicationPrograms> relevantProgramsForApplicant(
+      long applicantId, CiviFormProfile requesterProfile) {
     // Note: The Program model associated with the application is eagerly loaded.
     CompletableFuture<ImmutableSet<Application>> applicationsFuture =
         applicationRepository
             .getApplicationsForApplicant(
                 applicantId, ImmutableSet.of(LifecycleStage.DRAFT, LifecycleStage.ACTIVE))
             .toCompletableFuture();
+    ImmutableList<ProgramDefinition> activeProgramDefinitionsFuture =
```
[the remark this answers is not in the document]
Comment (swatkat1) [reply server/app/services/applicant/ApplicantService.java]: Done

File: server/app/services/applicant/ApplicantService.java [hunk 2]
```diff
@@ -834,39 +834,33 @@ public ImmutableList<Application> getApplications(TimeFilter submitTimeFilter) {
...
-                        .map(Program::getProgramDefinition)
-                        .filter(
-                            pdef ->
-                                pdef.displayMode().equals(DisplayMode.PUBLIC)
-                                    || (isTi && pdef.displayMode().equals(DisplayMode.TI_ONLY)))
-                        .collect(ImmutableList.toImmutableList()))
-            .toCompletableFuture();
-
-    return CompletableFuture.allOf(applicationsFuture, activeProgramDefinitionsFuture)
+    return CompletableFuture.allOf(applicationsFuture)
```
[the remark this answers is not in the document]
Comment (swatkat1) [reply server/app/services/applicant/ApplicantService.java]: updated

File: server/app/services/applicant/ApplicantService.java [hunk 3]
```diff
@@ -890,11 +889,12 @@ public CompletionStage<ApplicationPrograms> relevantProgramsForApplicant(long ap
    *     may be eligible for. Includes programs with matching eligibility criteria or no eligibility
    *     criteria.
    *     <p>Does not include the Common Intake Form.
-   *     <p>"Appropriate programs" those returned by {@link #relevantProgramsForApplicant(long)}.
+   *     <p>"Appropriate programs" those returned by {@link #relevantProgramsForApplicant(long,
+   *     play.mvc.Http.Request)}.
```
Comment (bion) [inline server/app/services/applicant/ApplicantService.java:893]: **nitpick (blocking):** looks like `relevantProgramsForApplicant` takes a `CiviFormProfile` not a `Http.Request`?

**Review by bion — APPROVED** [review]

File: server/app/services/applicant/ApplicantService.java (hunk 3 above)
Comment (swatkat1) [reply server/app/services/applicant/ApplicantService.java:893]: oh my! good catch will update it

**Review by bion — APPROVED** [review]

**Pull request merged**